src/extractor/guidance/turn_handler.cpp

```diff
@@ -773,17 +774,18 @@ void TurnHandler::handleDistinctConflict(const EdgeID via_edge,
         return;
     }
 
-    if (getTurnDirection(left.angle) == DirectionModifier::Right)
+    // turn to the right
+    if (getTurnDirection(left.angle) <= 180)
```

Comment:
this is the main bug here. By checking for the modifier, we do not use right turns when the angles involved are slight/sharp